.gitignore

```diff
@@ -39,3 +40,4 @@ callgrind.out.*
 tests/autotests/autotests
 .ycm_extra_conf.py*
 *.stash
+
```

Comment:
Do you have some link for your standard? The only standard I know of is to have newline at the end of the file (which is also what Git warns about if it isn't there), but I never heard of double newline at the end of file (and also I find it annoying).